birdhouse/pavics-compose.sh

```diff
@@ -85,30 +81,58 @@ for adir in $AUTODEPLOY_EXTRA_REPOS; do
 done
 export AUTODEPLOY_EXTRA_REPOS_AS_DOCKER_VOLUMES
 
-# we apply all the templates
-find $ALL_CONF_DIRS -name '*.template' |
-  while read FILE
-  do
-    DEST=${FILE%.template}
-    cat ${FILE} | envsubst "$VARS" | envsubst "$OPTIONAL_VARS" > ${DEST}
-  done
+BUILD_DIR="${BUILD_DIR:-"${COMPOSE_DIR}/build"}"
+
+COMPOSE_FILE="${BUILD_DIR}/docker-compose.yml"
+
+# Keep the compose project name as "birdhouse" by default (no matter where the build directory is located)
+export COMPOSE_PROJECT_NAME=${COMPOSE_PROJECT_NAME:-birdhouse}
 
 if [ x"$1" = x"up" ]; then
+  rm -r "${BUILD_DIR}"
   for adir in $ALL_CONF_DIRS; do
+    CONF_NAME="$(basename "${adir}")"
+    find "$adir" -type f |
+      while read -r FILE
+      do
+        RELATIVE_FILE_PATH=${FILE#${adir}}
+        DEST="${BUILD_DIR}/${CONF_NAME}/${RELATIVE_FILE_PATH#/}"
+        mkdir -p "$(dirname "${DEST}")"
+        ln "${FILE}" "${DEST}"
+      done
+  done
+
+  # we apply all the templates
+  find "$BUILD_DIR" -name '*.template' |
```

Comment:
Yeah! I didn't even think of that but yes that's a good point.

Reply:
So instead of the double search matches in `.template` and their instantiated files, we will now have 3 because of the duplicated template in the build dir?

Reply:
No, we only have the instantiated files in the build directory

Reply:
Yes

Reply:
Actually there is a 4th because the birdhouse directory has a symlink in `docs`

Reply:
For our dev machine, I think we can set `BUILD_DIR` outside of `birdhouse-deploy` checkout to avoid those duplicate files when searching.

Reply:
If `BUILD_DIR=birdhouse` is supported, then this is a viable option for a dev machine.

Reply:
Currently, this is not supported since the arrangement of files in the `BUILD_DIR` is not the same as the one in `birdhouse`. 

But... if we moved all of the components, from `components`, `config`, `optional-components`, etc. into one single location that could be possible. If we did that though, you'd have to also put any additional components into that directory as well.

For me, I would just adjust my search filters to not read from the `BUILD_DIR` (or make it outside of the project folder like @tlvu suggests). Then if I explicitly want to search in the `BUILD_DIR` I can select that folder to search. This is what I currently do to avoid looking in the `docs/source/birdhouse/` directory for duplicates.

I mostly use PyCharm and VSCode and it's pretty easy to set up these filters, I can demo it for you if you'd like. I'm sure most other IDEs support a similar search filtering.